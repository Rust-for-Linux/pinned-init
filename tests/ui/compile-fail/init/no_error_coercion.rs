#![feature(allocator_api)]
use pin_init::*;
use std::alloc::AllocError;

struct Foo {
    a: Box<usize>,
    bar: Bar,
}

struct Bar {
    b: usize,
}

impl Foo {
    fn new() -> impl Init<Self, AllocError> {
        try_init!(Self {
            a: Box::new(42),
            bar <- init!(Bar { b: 42 }),
        }? AllocError)
    }
}

fn main() {}
