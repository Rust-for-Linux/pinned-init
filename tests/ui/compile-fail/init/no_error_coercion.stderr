error[E0277]: `?` couldn't convert the error to `std::alloc::AllocError`
  --> tests/ui/compile-fail/init/no_error_coercion.rs:19:22
   |
16 | /         try_init!(Self {
17 | |             a: Box::new(42),
18 | |             bar <- init!(Bar { b: 42 }),
19 | |         }? AllocError)
   | |                      ^
   | |                      |
   | |______________________the trait `From<Infallible>` is not implemented for `std::alloc::AllocError`
   |                        this can't be annotated with `?` because it has type `Result<_, Infallible>`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = note: this error originates in the macro `try_init` (in Nightly builds, run with -Z macro-backtrace for more info)
